src/app/_services/depublication.service.ts

```diff
@@ -0,0 +1,129 @@
+import { HttpClient, HttpEvent, HttpEventType } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { apiSettings } from '../../environments/apisettings';
+import { MoreResults, RecordPublicationInfo, Results, SortParameter } from '../_models';
+
+import { collectResultsUptoPage } from './service-utils';
+import { ErrorService } from './error.service';
+
+@Injectable({ providedIn: 'root' })
+export class DepublicationService {
+  constructor(private readonly http: HttpClient, private readonly errors: ErrorService) {}
+
+  /** handleUploadEvents
+  /*  record upload events
+  /*  @param {HttpEvent} event - event from the server
+  */
+  handleUploadEvents(event: HttpEvent<Object>): boolean {
```

Comment:
According to my IDE, this can be a static function. Is that generally recommended?
Note: same for the functions `parseFilterParameter` and `parseSortParameter` below.

Reply:
I assume unit tests are an issue here too?